### PR TITLE
[web3torrent] Add some notes on how channels are integrated

### PR DESCRIPTION
**Pull request by geoknee**
Theses notes should go into greater detail than the readme, and will provide a source of reference for devs and for a future blog post. 

At this time I don't think the code is sufficiently clear as to be self documenting, so I think this will be useful. 

PR also moves some constants to `constants.ts`.

### CODE REVIEW

File: packages/web3torrent/notes/channels-integration.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+These tradeoffs are set in [`constants.ts`](https://github.com/statechannels/monorepo/blob/master/packages/web3torrent/src/constants.ts).
+
+## Payment channel client
+
+Web3Torrent includes a general purpose, 2-party unidirectional payment channel client that wraps the state channels wallet provider and provides convenient methods such as `sendPayment`. This client takes care of all downstream encoding to interface with the state channels stack.
+
+## Paid streaming extension
+
+Web3Torrent consists of a paid streaming extension to the webTorrent protocol. The purpose of the extension is
```
Comment (junjizhi) [inline packages/web3torrent/notes/channels-integration.md]: I think the precise term should be an extension of [BitTorrent protocol](https://webtorrent.io/docs).

File: packages/web3torrent/notes/channels-integration.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+- to keep track of peers that have requested data from me, and to maintain a buffer representing the number of bytes I am willing to upload to them
+- to interface with the state channels wallet in order to setup and teardown payment channels with my peers
+- to send and recieve crypto-money in those payment channels in exchange for data
```
Comment (junjizhi) [inline packages/web3torrent/notes/channels-integration.md]: The current implementation is the extension is simply emitting the events up a layer and let the torrent do all the peer tracking and channel interfacing work. Should we just rename this section `Paid streaming extension` to `Web3Torrent paid streaming client`.

File: packages/web3torrent/notes/channels-integration.md (hunk 1 above)
Comment (geoknee) [reply packages/web3torrent/notes/channels-integration.md]: Hopefully clearer in 91fad3d

File: packages/web3torrent/notes/channels-integration.md (hunk 2 above)
Comment (geoknee) [reply packages/web3torrent/notes/channels-integration.md]: 91fad3d

File: packages/web3torrent/notes/channels-integration.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
+# How are channels integrated into web3Torrent?
+
+Web3Torrent uses state channels tech to stream money between webTorrent peers, so that downloaders may pay uploaders for data. The granularity of the payments can be tuned, which alters the trust-performance tradeoff. For example, peers can pay per byte (low trust, high overhead for processing payments resulting in low data transfer speeds) or per Megabyte (higher trust, lower overhead and greater transfer speed).
+
+These tradeoffs are set in [`constants.ts`](https://github.com/statechannels/monorepo/blob/master/packages/web3torrent/src/constants.ts).
+
+## Paid streaming extension
+
+We wrote a small extension to the BiTorrent protocol that allows for extra information to exchanged between torrenting peers: sufficient for them to run a state channel.
```
Comment (spersico) [inline packages/web3torrent/notes/channels-integration.md]: It should also be mentioned that this extension is the one in charge of intercepting and blocking and resuming data transfers, to control the flow of data between two peers.

File: packages/web3torrent/src/library/web3torrent-lib.ts [hunk 4]
```diff
@@ -28,27 +37,7 @@ export type TorrentCallback = (torrent: Torrent) => any;
 
 export * from './types';
 
-export const WEI_PER_BYTE = bigNumberify(1); // cost per byte
```
Comment (spersico) [inline packages/web3torrent/src/library/web3torrent-lib.ts]: :+1:

File: packages/web3torrent/notes/channels-integration.md [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+In more detail:
+
+### Messaging system
+
+When the `MessageQueued` event fires, send the associated data down the `wire`. When a message is received on the wire, pass it on to the state channels wallet using `pushMessage`.
+
+### Channel setup
+
+Uploader: Upon the first request for data of an unkown peer, choke them and send them a `STOP` notification (which is a prompt for a payment). Also create a new, unidirectional payment channel with me (the uploader) as the "beneficiary", and the peer (the downloader) as the "payer". This action is associated with the `paymentChannelClient.createChannel()` API call.
```
Comment (spersico) [inline packages/web3torrent/notes/channels-integration.md]: I think the order of events here is wrong, The unidirectional payment channel is created, and *then* the uploader chokes them and sends a `STOP` notification.

File: packages/web3torrent/notes/channels-integration.md [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+
+Downloader: When the "ChannelProposed" event fires, automatically join that channel.
+
+### Payments
+
+Uploader: if a payment is received, "accept" it.
```
Comment (spersico) [inline packages/web3torrent/notes/channels-integration.md]: ... and unchoke the downloader, and send a `START` notification

**Comment by geoknee** [comment]
Thanks for the feedback @junjizhi  and @spersico . I will address outstanding comments and merge this later on.

**Pull request merged**